### ISSUE TITLE
Theia fails to load with webview and preferences opened

### ISSUE THREAD

**Issue by shahar-h** [eclipse-theia/theia#6877]
### Description

When refreshing browser while some Webview widget is opened in the **first** tab, and `preferences` in the **second** tab, Theia fails to load without any error in browser console:
![image](https://user-images.githubusercontent.com/16555289/72280260-9ee78000-3640-11ea-978f-b626c51bdfe2.png)
After clearing `layout` value from browser local storage Theia is able to load again.

### Reproduction Steps
1. Open some Webview widget. I loaded [GitLens](https://marketplace.visualstudio.com/items?itemName=eamodio.gitlens) vscode extension and then opened its settings with `GitLens: Open Settings` comand.
2. Open preferences in a second tab: `File->Settings->Open Preferences`
3. Refresh browser.

**OS and Theia version:**
Windows 10.
Theia master.

**Diagnostics:** 
By debugging the flow I noticed that the issue resides in `restoreLayout` method inside `shell-layout-restorer`: `await app.shell.setLayoutData(layoutData);` call is not resolved.
https://github.com/eclipse-theia/theia/blob/f11045a0f886b3a6ad10fbc5255288abff424aeb/packages/core/src/browser/shell/shell-layout-restorer.ts#L170

**Comment by akosyakov** [comment]
Which Theia and VS Code extensions you have installed?

**Comment by shahar-h** [comment]
I built Theia from master branch and loaded [GitLens](https://marketplace.visualstudio.com/items?itemName=eamodio.gitlens) vscode extension, but I could reproduce this issue with any webview.

**Comment by vince-fugnitto** [comment]
I think its related to https://github.com/eclipse-theia/theia/issues/6287 also.

**Comment by amiramw** [comment]
@akosyakov do you know if there is any work on that issue currently or we can take it?

**Comment by vince-fugnitto** [comment]
> @akosyakov do you know if there is any work on that issue currently or we can take it?

I don't think anyone is working on the issue, we'd be happy for you to take it @amiramw.

**Comment by DoroNahari** [comment]
I checked this issue again.
When repeating the steps written above with git lens webview, Theia is loaded but the editor area is blank/ lost it's state. i didn't saw any error or warning in the browser console.
I checked with other random webviews and some have the same issue and some have no issue at all.

issue (git lens):
![ezgif com-video-to-gif (2)](https://user-images.githubusercontent.com/24614792/84865754-6598bd80-b081-11ea-81c4-d538963d9310.gif)

no issue (svg viewer):
![ezgif com-video-to-gif (1)](https://user-images.githubusercontent.com/24614792/84865525-063aad80-b081-11ea-9b50-d5330292c66f.gif)
